### PR TITLE
DM-22531: AuxTel on-sky testing Dec/2019

### PR DESCRIPTION
**Pull request by tribeiro**
Add notebooks developed during operations. (+4 squashed commits)
Squashed commits:
[5a229ab] Add quick rotator test script.
[0b6239d] Add Basic Rotator test notebook
[da73e77] Add end-to-end verification notebook. Add ATSpectrograph notebook.
[68e53be] Add example notebook with how to get data from influxdb EFD

### CODE REVIEW

File: examples/AT start up.ipynb [hunk 1]
```diff
@@ -0,0 +1,422 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook contains a template for basic startup operations for the Auxiliary telescope. At the begging of the night a user can run the startup procedure to prepare the telescope for observing. Omce it is done, remember to shutdown the notebook server. \n",
```
Comment (patrickingraham) [inline examples/AT start up.ipynb]: Typos: 
beginning, once

File: examples/AT start up.ipynb [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+   "metadata": {},
+   "source": [
+    "Setup remotes for all the AT components"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "%%script bash \n",
```
Comment (patrickingraham) [inline examples/AT start up.ipynb]: Maybe just explain when it is required to do this. Maybe put it in a separate cell and say, "uncomment and run if IDL files don't exist" or something like that.

File: examples/Go2Target.ipynb [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+   "metadata": {},
+   "source": [
+    "# ATMCS"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If the ATMCS is not enable run the next cell."
```
Comment (patrickingraham) [inline examples/Go2Target.ipynb]: enabled,

File: examples/Go2Target.ipynb [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+   "metadata": {},
+   "source": [
+    "# Going to the Target"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The next cell sets the observatory location. This is needed to compute the Az/El of the target to set the camera rotation angle. We are trying to keep the angle close to zero. "
```
Comment (patrickingraham) [inline examples/Go2Target.ipynb]: which angle? no obvious from the next cell....

File: examples/Go2Target.ipynb [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+    "    equinox=2000,  # should be ignored for ICRS\n",
+    "    ra=radec.ra.hour,\n",
+    "    declination=radec.dec.deg,\n",
+    "    parallax=0,\n",
+    "    pmRA=0,\n",
+    "    pmDec=0,\n",
+    "    rv=0,\n",
+    "    dRA=0,\n",
+    "    dDec=0,\n",
+    "    rotPA=180.-alt_az.alt.deg,\n",
```
Comment (patrickingraham) [inline examples/Go2Target.ipynb]: Say this is where the rotator position is optimized to be near zero. Also say what would be the default value if you weren't trying to set it near zero

File: examples/Go2Target.ipynb [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+    "await atptg.cmd_offsetAzEl.set_start(el=0., \n",
+    "                                     az=-100., \n",
+    "                                         num=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you want to make persistet offsets you can use the following method. "
```
Comment (patrickingraham) [inline examples/Go2Target.ipynb]: I don't really understand this. What do you mean by persistent and when would I want to do that?

File: examples/Go2Target.ipynb [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+    "await atptg.cmd_offsetAzEl.set_start(el=0., \n",
+    "                                     az=-100., \n",
+    "                                     num=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you want to add your offset to a pointing model file, do the following. "
```
Comment (patrickingraham) [inline examples/Go2Target.ipynb]: Is this really part of a standard slew? It feels outside the scope of a slew example. Might belong in a "create a pointing model" example?

File: examples/Go2Target.ipynb [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await attcs.slew(ra=\"20:25:38.85705\", \n",
```
Comment (patrickingraham) [inline examples/Go2Target.ipynb]: why is this here?

File: examples/Make_Grid.ipynb [hunk 9]
```diff
@@ -0,0 +1,444 @@
...
+    "from lsst.ts.idl.enums import ATPtg\n",
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Disable auto download of iers data. "
```
Comment (patrickingraham) [inline examples/Make_Grid.ipynb:69]: One might say  what iers data is in brackets....

File: examples/Make_Grid.ipynb [hunk 10]
```diff
@@ -0,0 +1,444 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "While the pointing component does not support the slew and track of an alt/az position, the following cells will provide a quick and easy way to perform this task. The user specified a position in Alt/Az and it will use astropy coordinate library to convert it to RA/Dec. \n",
+    "\n",
+    "For that we will need the location of the observatory and the time. Location is defined as an astropy `EarthLocation` and time is taken from the pointing component and then creating an astropy `Time` object.\n",
+    "\n",
+    "Obervatory location (lon, lat and height extracted from: https://github.com/lsst/sims_utils)"
```
Comment (patrickingraham) [inline examples/Make_Grid.ipynb]: *Observatory

File: examples/Make_Grid.ipynb [hunk 11]
```diff
@@ -0,0 +1,444 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await asyncio.sleep(5.)\n",
+    "\n",
+    "for i in range(10):\n",
```
Comment (patrickingraham) [inline examples/Make_Grid.ipynb]: I must be really daft but I don't see how this is changing targets. I don't see any offset that is iterated over...

File: examples/Make_Grid.ipynb [hunk 12]
```diff
@@ -0,0 +1,444 @@
...
+    "await atptg.cmd_stopTracking.start(timeout=10.)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await atptg.cmd_azElTarget.set_start(azDegs=0., elDegs=80.)"
```
Comment (patrickingraham) [inline examples/Make_Grid.ipynb]: Why is this here? it parks the telescope...

**Review by patrickingraham — CHANGES_REQUESTED** [review]

**Comment by tribeiro** [comment]
I think I covered the last comments on the PR.

**Review by patrickingraham — APPROVED** [review]
Looks much better!

**Pull request merged**